main/FatesRestartInterfaceMod.F90

```diff
@@ -2461,8 +2534,11 @@ subroutine set_restart_vectors(this,nc,nsites,sites)
           end do
           
           do i = 1,numWaterMem ! numWaterMem currently 10
-             rio_watermem_siwm( io_idx_si_wmem ) = sites(s)%water_memory(i)
-             io_idx_si_wmem = io_idx_si_wmem + 1
+             do i_pft=1,numpft
```

Comment:
Because of this, we need to make sure enough restart memory is allocated, see: https://github.com/NGEET/fates/blob/sci.1.65.7_api.25.4.0/main/FatesInterfaceMod.F90#L835

Should update it to: 

```
fates_maxElementsPerSite = max(fates_maxPatchesPerSite * fates_maxElementsPerPatch, &
              numWatermem*numpft, num_vegtemp_mem, num_elements, nlevsclass*numpft)
```